src/Symfony/Component/DependencyInjection/ContainerBuilder.php

```diff
@@ -1194,6 +1196,40 @@ public static function getServiceConditionals($value)
     }
 
     /**
+     * Checks whether the requested file or directory exists and registers the result for resource tracking.
+     *
+     * @param string      $path          The file or directory path for which to check the existence
+     * @param bool|string $trackContents Whether to track contents of the given resource. If a string is passed,
+     *                                   it will be used as pattern for tracking contents of the requested directory
+     *
+     * @return bool
+     *
+     * @final
+     */
+    public function fileExists($path, $trackContents = true)
+    {
+        $exists = file_exists($path);
+
+        if (!$this->trackResources) {
+            return $exists;
+        }
+
+        if (!$exists) {
+            $this->addResource(new FileExistenceResource($path));
+        }
+
+        if ($trackContents) {
+            if (is_file($path)) {
+                $this->addResource(new FileResource($path));
+            } elseif (is_dir($path)) {
```

Comment:
what but just an "else" here? otherwise, it looks strange to has a potential case were we don't do anything (the implicit else here)